File: ios/Classes/AnymexExtensionRuntimeBridgePlugin.m
#import "AnymexExtensionRuntimeBridgePlugin.h"

@implementation AnymexExtensionRuntimeBridgePlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel =
      [FlutterMethodChannel methodChannelWithName:@"anymex_extension_runtime_bridge"
                                  binaryMessenger:[registrar messenger]];
  AnymexExtensionRuntimeBridgePlugin* instance = [[AnymexExtensionRuntimeBridgePlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    NSString* version = [@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]];
    result(version);
    return;
  }
  result(FlutterMethodNotImplemented);
}

@end

File: ios/Classes/AnymexExtensionRuntimeBridgePlugin.swift
import Flutter
import UIKit

public class AnymexExtensionRuntimeBridgePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "anymex_extension_runtime_bridge", binaryMessenger: registrar.messenger())
    let instance = AnymexExtensionRuntimeBridgePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
